spyderlib/restart_app.py

```diff
@@ -128,19 +186,52 @@ def main():
     # Adjust the command and/or arguments to subprocess depending on the OS
     shell = os.name != 'nt'
 
-    # Wait for original process to end before launching the new instance
-    while True:
+    # Before launching a new Spyder instance we need to make sure that the
+    # previous one has closed. We wait for a fixed and "reasonable" amount of
+    # time and check, otherwise an error is launched
+    wait_time = 60*5  # Seconds

```

Comment:
1. I guess, but windows sometimes takes sooooooo long!
2. Sure we could, what would you like to see? a "splash screen saying, Spyder is restarting and the throbber?